pyuvdata/uvdata/uvdata.py

```diff
@@ -241,11 +243,14 @@ def __init__(self):
             expected_type=np.float,
             tols=1e-3,
         )  # 1 ms
+
+        desc = (
+            "Width of frequency channels (Hz). If flex_spw = False, then it single "
+            "value of type = float, otherwise it is an array of shape (Nfreqs), type = "
+            "float."
+        )
         self._channel_width = uvp.UVParameter(
-            "channel_width",
-            description="Width of frequency channels (Hz)",
-            expected_type=np.float,
-            tols=1e-3,
+            "channel_width", description=desc, expected_type=np.float, tols=1e-3,
         )  # 1 mHz
```

Comment:
I'm certainly happy to just force this to be an array -- while implementing the changes, I was attempting to minimize the potential for any breaking changes, which is why I allowed it to remain just a solitary float. The only place that one would need to take care to wrap this would be w/ older UDH5 files, because of course this is effectively encoding a change to the file structure. I do already have a catch build in for `read_uvh5`, so it wouldn't be difficult to expand that just a tad (all that to say: "I'm happy to go with whatever makes you guys happy").

Reply:
I've scheduled a conversation with HERA folks, I'll get back to you after that about this. Your approach is safer, I agree, so we'll probably stick with that for this PR.